### PR TITLE
ci: check scripts

### LINKED ISSUE #917: ci: check whether scripts still compile
### Clear and concise description of the problem

In #910 I removed code from the sources, that were used in the scripts.
Unfortunately, there isn't a build step that checks them with TS powers.

### Suggested solution

Check the types in the ci pipeline.

### Alternative

Run all scripts (#813), however that still wouldn't catch most type errors.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #917

### CODE REVIEW

**Comment by Shinigami92** [comment]
TS-Errors are reported successfully:

**Comment by Shinigami92** [comment]
awaiting merge of #915

**Comment by Shinigami92** [comment]
I don't know why but the results differs when I run
`pnpm run ts-check:scripts` vs `tsc --noEmit --esModuleInterop --skipLibCheck scripts/**/*.ts` manually 🤔

**Comment by ST-DDT** [comment]
This looks like the perfect place for using [extends](https://www.typescriptlang.org/tsconfig#extends).

**Comment by Shinigami92** [comment]
> This looks like the perfect place for using [extends](https://www.typescriptlang.org/tsconfig#extends).

I didn't use it for `tsconfig.json` as I'm to afraid to break something in build process.
Also I needed to reverse some settings explicit set for scripts and tests and therefore the file is nearly same length 🤷

**Comment by ST-DDT** [comment]
#915 is merged.

File: package.json [hunk 1]
```diff
@@ -110,6 +112,7 @@
     "eslint-plugin-jsdoc": "~39.2.9",
     "eslint-plugin-prettier": "~4.0.0",
     "esno": "~0.14.1",
+    "glob": "~8.0.1",
```
Comment (ST-DDT) [inline package.json:115]: Why did you add this one?

**Review by ST-DDT — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Shinigami92) [reply package.json:115]: https://github.com/faker-js/faker/blob/b4cadfaae924218ae605c94f91e1206990d818d9/scripts/bundle.ts#L2

Comment (ST-DDT) [reply package.json:115]: That file was not edited by this PR.

Comment (Shinigami92) [reply package.json:115]: It was not explicitly in dependencies 
Now it checks the ts code and needs it as direct dependency 
Check the main branch and look into this file, you will see red squiggles but previously they could be ignored because one of our dependencies already depend on glob
Use `pnpm why glob` in main to find out which one

**Review by import-brain — APPROVED** [review]
